src/Agent.Listener/Configuration/AutoLogonManager.cs

```diff
@@ -97,6 +97,13 @@ public async Task ConfigureAsync(CommandSettings command)
                 Trace.Info($"Continuing with the autologon configuration.");
             }
 
+            // grant permssion for agent root folder and work folder
+            Trace.Info("Create local group and grant folder permission to logon account.");
```

Comment:
This part of the code looks similar to the one that's under `WindowsServiceControlManager.cs`, around line 107. Consider extracting this into a separate method, or alternatively if we only expect to grant permissions to well-known directories, move it under `GrantDirectoryPermissionForAccount()`.